pallets/storage-bench/src/lib.rs

```diff
@@ -0,0 +1,390 @@
+            let (utxo, encrypted_note) = ShardsTwoxIdentity::<T>::get(shard_idx, encoded_utxo_idx);
+            Self::deposit_event(Event::ShardElementRead(utxo, encrypted_note));
+            Ok(())
+        }
+
+        #[pallet::weight(500_000)]
+        #[transactional]
+        pub fn point_read_shard_element_twox_twox(
+            origin: OriginFor<T>,
+            shard_idx: u8,
+            utxo_idx: u64,
+        ) -> DispatchResult {
+            T::UpdateOrigin::ensure_origin(origin)?;
+            let (utxo, encrypted_note) = ShardsTwoxTwox::<T>::get(shard_idx, utxo_idx);
+            Self::deposit_event(Event::ShardElementRead(utxo, encrypted_note));
+            Ok(())
+        }
+
+        #[pallet::weight(500_000)]
+        pub fn range_read_shard_element_identity_identity(
+            origin: OriginFor<T>,
+            shard_idx: u8,
+            utxo_idx: u64,
+            amount: u64,
+        ) -> DispatchResult {
+            T::UpdateOrigin::ensure_origin(origin)?;
+            let mut receivers = Vec::new();
+            let mut iter = if utxo_idx == 0 {
+                ShardsIdentityIdentity::<T>::iter_prefix(shard_idx)
+            } else {
+                let raw_key = ShardsIdentityIdentity::<T>::hashed_key_for(
+                    shard_idx,
+                    Self::encode(utxo_idx as u64 - 1),
+                );
+                ShardsIdentityIdentity::<T>::iter_prefix_from(shard_idx, raw_key)
+            };
+            for _ in 0..amount {
+                match iter.next() {
+                    Some((_, next)) => receivers.push(next),
+                    _ => return Err(Error::<T>::NullStorageItem.into()),
+                }
+            }
+            Self::deposit_event(Event::ShardElementBatchRead(amount));
+            Ok(())
+        }
+
+        #[pallet::weight(500_000)]
+        pub fn range_read_shard_element_twox_identity(
+            origin: OriginFor<T>,
+            shard_idx: u8,
+            utxo_idx: u64,
+            amount: u64,
+        ) -> DispatchResult {
+            T::UpdateOrigin::ensure_origin(origin)?;
+            let mut receivers = Vec::new();
+            let mut iter = if utxo_idx == 0 {
+                ShardsTwoxIdentity::<T>::iter_prefix(shard_idx)
+            } else {
+                let raw_key = ShardsTwoxIdentity::<T>::hashed_key_for(
+                    shard_idx,
+                    Self::encode(utxo_idx as u64 - 1),
+                );
+                ShardsTwoxIdentity::<T>::iter_prefix_from(shard_idx, raw_key)
+            };
+            for _ in 0..amount {
+                match iter.next() {
+                    Some((_, next)) => receivers.push(next),
+                    _ => return Err(Error::<T>::NullStorageItem.into()),
+                }
+            }
+            Self::deposit_event(Event::ShardElementBatchRead(amount));
+            Ok(())
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        fn encode(n: u64) -> u64 {
```

Comment:
This is a small function and called in several places, `#[inline]` would be better.